scylla/src/transport/topology.rs

```diff
@@ -463,7 +464,7 @@ impl MetadataReader {
         let control_connection_peer = metadata
             .peers
             .iter()
-            .find(|peer| peer.address == self.control_connection_endpoint.address());
+            .find(|peer| peer.address.into_inner() == self.control_connection_endpoint.address());
```

Comment:
I think this is wrong. We shouldn't compare SocketAddrs but rather NodeAddrs. We cannot consider `RealPeer(x)` to be equal to `FixedLocal(x)` for any `x`.